static/app/components/smartSearchBar/searchBarDatePicker.tsx

```diff
@@ -0,0 +1,214 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+import format from 'date-fns/format';
+import type {Moment} from 'moment';
+
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {DEFAULT_DAY_START_TIME, setDateToTime} from 'sentry/utils/dates';
+
+import {DatePicker} from '../calendar';
+import Checkbox from '../checkbox';
+
+import SearchBarFlyout from './searchBarFlyout';
+
+type SearchBarDatePickerProps = {
+  handleSelectDateTime: (value: string) => void;
+  date?: Moment;
+  dateString?: string;
+};
+
+type TimeInputProps = {
+  setTime: (newTime: string) => void;
+  time: string;
+};
+
+const TZ_OFFSET_REGEX = /[+-]\d\d:\d\d$/;
+const TIME_REGEX = /T\d\d:\d\d:\d\d/;
+
+const ISO_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
+const ISO_FORMAT_WITH_TIMEZONE = ISO_FORMAT + 'xxx';
+
+const isUtcIsoDate = (isoDateTime?: string) => {
+  if (!isoDateTime) {
+    return false;
+  }
+
+  return !TZ_OFFSET_REGEX.test(isoDateTime);
+};
+
+const applyChanges = ({
+  date = new Date(),
```

Comment:
Do you know when this value is constructed? Is it at call time or module load time? I actually don't remember how javascript does this (vs like python, where it happens at module load time, so if you mutate the parameter it changes for all future calls lol)

Reply:
At call time, double checked in the repl

![image](https://user-images.githubusercontent.com/10888943/184029732-a8a8bd91-385b-4bf9-a13c-9f82e5818d8c.png)